### DISCUSSION TITLE
CLI command to restart Rectangle

### DISCUSSION THREAD

**Discussion by neutonfoo** [rxhanson/Rectangle#1027] [Q&A] [answered]
First of all, thanks for all the hard work that goes into this!

This is a bit of a niche use case, but I was wondering if there was a way to restart Rectangle from the CLI without having the preferences window open. I have a script that sets the "[Setting gaps at the screen edges](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#setting-gaps-at-the-screen-edges)" terminal command depending on if a monitor is my main display or not, but realize that these preferences aren't set until I restart Rectangle.

**Comment by rxhanson** [comment, marked as answer]
Is this what you're looking for?

```
osascript -e 'quit app "Rectangle"'
open -a Rectangle
```

https://apple.stackexchange.com/questions/354954/how-can-i-quit-an-app-using-terminal

If anyone is up for just building display dependent screen edge gaps into Rectangle, I see no reason to turn down a good pull request for it.

**Reply by neutonfoo** [reply]
Thanks for the quick reply! That's what I'm currently doing, but when I run `open -a Rectangle`, the preferences window opens. Is there a way to open Rectangle and not display the Preferences window? The `-g and -j` options don't seem to do the job unfortunately.

**Reply by rxhanson** [reply]
If the prefs window opens, that means you're launching it while it's already open. You might need a little delay?

**Reply by neutonfoo** [reply]
Ah wow I never realized that was the case. It turns out you need quotes around the app name in `osascript -e 'quit app "Rectangle"'`. Now it works without a delay. Thanks for your help!

**Reply by rxhanson** [reply]
Good to know, thanks! I updated my answer here to reflect that.
